docroot/modules/custom/foia_request/foia_request.drush.inc

```diff
@@ -0,0 +1,158 @@
+<?php
+
+/**
+ * @file
+ * Drush command to update FOIA request submissions.
+ */
+
+use Drupal\webform\entity\WebformSubmission;
+use Drupal\file\Entity\File;
+use Drupal\foia_request\Entity\FoiaRequestInterface;
+use Drupal\foia_request\Entity\FoiaRequest;
+
+/**
+ * Implements hook_drush_command().
+ */
+function foia_request_drush_command() {
+  $commands = [];
+
+  $commands['queue-clean-request'] = [
+    'description' => 'Queue clean FOIA request when files have been scanned.',
+    'aliases' => ['qcr'],
+  ];
+  return $commands;
+}
+
+/**
+ * Implements hook_drush_help().
+ */
+function foia_request_drush_help($section) {
+  switch ($section) {
+    case 'update-foia-request':
+      return dt('This command will update the status of a FOIA request if all the files have been scanned.');
+  }
+}
+
+/**
+ * Implements drush_hook_COMMAND().
+ */
+function drush_foia_request_queue_clean_request() {
+  $requests_to_be_scanned = foia_request_to_be_scanned();
+  if (!empty($requests_to_be_scanned)) {
+    foia_request_check_request($requests_to_be_scanned);
+  }
+}
+
+/**
+ * Check each request to see if it can be queued.
+ *
+ * @param array $requests_to_be_scanned
+ *   An array containing FOIA Requests IDs to be checked.
+ */
+function foia_request_check_request(array $requests_to_be_scanned) {
+  foreach ($requests_to_be_scanned as $request) {
+    $ready_to_queue = foia_request_check_file_fields($request);
+    if ($ready_to_queue) {
+      foia_request_queue_request($request);
+    }
+  }
+}
+
+/**
+ * Checks files for every file field on a given request.
+ *
+ * @param object $request
+ *   An database object containing the submission ID and foia_request ID.
+ *
+ * @return bool
+ *   Returns TRUE if all of the files have been scanned, else FALSE.
+ */
+function foia_request_check_file_fields($request) {
+  $attachment_fields = &drupal_static(__FUNCTION__);
+  $webform_submission = WebformSubmission::load($request->sid);
+
+  if (isset($attachment_fields[$request->webform_id])) {
+    $file_fields = $attachment_fields[$request->webform_id];
+  }
+  else {
+    $webform = $webform_submission->getWebform();
+    $file_field_lookup = \Drupal::service('foia_webform.file_field_lookup_service');
```

Comment:
Should we use one of the preexisting getFileAttachmentElementsOnWebform protected functions from the FoiaSubmissionServiceApi or FoiaSubmissionQueueHandler classes instead of creating a new lookup service for fields?

Reply:
The intent is to remove duplicate code and turn it into a service. We should refactor and remove the methods from the other classes and use the service.